api/utils/clientutils/resources.go

```diff
@@ -28,26 +29,60 @@ import (
 
 // IterateResources is a helper that iterates through each resource from all
 // pages and passes them one by one to the provided callback.
+// Deprecated: Prefer using [Resources] instead.
+// TODO(tross): DELETE IN 19.0.0
 func IterateResources[T any](
 	ctx context.Context,
-	listPageFunc func(context.Context, int, string) ([]T, string, error),
+	pageFunc func(context.Context, int, string) ([]T, string, error),
 	callback func(T) error,
 ) error {
-	var pageToken string
-	for {
-		page, nextToken, err := listPageFunc(ctx, defaults.DefaultChunkSize, pageToken)
+	for item, err := range Resources(ctx, pageFunc) {
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		for _, resource := range page {
-			if err := callback(resource); err != nil {
-				return trace.Wrap(err)
-			}
+
+		if err := callback(item); err != nil {
+			return trace.Wrap(err)
 		}
+	}
+
+	return nil
+}
 
-		if nextToken == "" {
-			return nil
+// Resources returns an iterator over all resources from every page produced from the pageFunc.
+// The iterator will only produce an error if one is encountered retrieving a page.
+func Resources[T any](ctx context.Context, pageFunc func(context.Context, int, string) ([]T, string, error)) iter.Seq2[T, error] {
+	return func(yield func(T, error) bool) {
+		var pageToken string
+		limit := defaults.DefaultChunkSize
```

Comment:
The "limit" name suggests that the iteration will stop after that many items, but it's just a page size.